rd_ui/app/scripts/controllers/query_view.js

```diff
@@ -230,12 +230,20 @@
       $scope.$parent.pageTitle = $scope.query.name;
     });
 
+    var firstLoad = true;
     $scope.$watch('queryResult && queryResult.getData()', function(data, oldData) {
       if (!data) {
         return;
       }
 
       $scope.filters = $scope.queryResult.getFilters();
+      if (firstLoad) {
+        _.each($scope.filters, function (queryFilter) {
+          if (_.has($location.search(), queryFilter.name))
+            queryFilter.current = $location.search()[queryFilter.name];
```

Comment:
If we're reading values from the URL, we should also write them to the URL to keep them in sync.
I think we have this functionality on dashboards - maybe we can share the code?